src/main/java/net/sf/jabref/model/database/BibDatabaseMode.java

```diff
@@ -2,34 +2,69 @@
 
 import java.util.Locale;
 
+/**
+ * An enum which contains the possible {@link BibDatabase} Modes.
+ * Possible are BibTeX and BibLaTeX.
+ */
 public enum BibDatabaseMode {
     BIBTEX,
     BIBLATEX;
 
+    /**
+     * @return the name of the current mode as String
+     */
     public String getFormattedName() {
-        if(this == BIBTEX) {
+        if (this == BIBTEX) {
             return "BibTeX";
         } else {
             return "BibLaTeX";
         }
     }
 
+    /**
+     * Returns the opposite mode of the current mode as {@link BibDatabaseMode}.
+     *
+     * @return BibLaTeX if the current mode is BIBTEX, BibTeX else
+     */
     public BibDatabaseMode getOppositeMode() {
-        if(this == BIBTEX) {
+        if (this == BIBTEX) {
             return BIBLATEX;
         } else {
             return BIBTEX;
         }
     }
 
+    /**
+     * Returns the {@link BibDatabaseMode} from a given boolean.
+     *
+     * @param isBibLatex
+     * @return BIBLATEX if isBibLatex is true, else BIBTEX
+     */
     public static BibDatabaseMode fromPreference(boolean isBibLatex) {
-        return isBibLatex ? BIBLATEX : BIBTEX;
+        if (isBibLatex) {
+            return BIBLATEX;
+        } else {
+            return BIBTEX;
+        }
     }
 
+    /**
+     * Returns the {@link BibDatabaseMode} that equals the given string.
+     * The use of capital and small letters in the string doesn't matter.
+     * If neither "bibtex" nor "biblatex" is the given string, then this will
```

Comment:
This comment in unclear and reveals some internal details. What is the consequence if the string is neither bibtex nor biblatex? Will it throw a RuntimeException?